Unable to build with latest ESP-IDF due to pthread_rwlock

It seems that ESP-IDF commit https://github.com/espressif/esp-idf/commit/b117bcd2a0a05b1dd334a5eb52a465ecfe6e01d1, has broken compilation on both ESP-IDF master and release/v4.4 branches, resulting in the below error:
```
  = note: Running ldproxy
          Error: Linker /home/user/.espressif/tools/xtensa-esp32s3-elf/esp-2021r2-8.4.0/xtensa-esp32s3-elf/bin/xtensa-esp32s3-elf-gcc failed: exit status: 1
          STDERR OUTPUT:
          /home/user/.espressif/tools/xtensa-esp32s3-elf/esp-2021r2-8.4.0/xtensa-esp32s3-elf/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld: esp-idf/pthread/libpthread.a(pthread_rwlock.c.obj): in function `pthread_rwlock_init':
          /home/user/.espressif/esp-idf-master/components/pthread/pthread_rwlock.c:51: multiple definition of `pthread_rwlock_init'; /home/user/projects/rust-esp32-std-demo/target/xtensa-esp32s3-espidf/release/deps/libesp_idf_sys-033449d0383ac0dd.rlib(esp_idf_sys-033449d0383ac0dd.esp_idf_sys.208449cf-cgu.8.rcgu.o):/home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/esp-idf-sys-0.24.4/src/pthread_rwlock.rs:15: first defined here
          /home/user/.espressif/tools/xtensa-esp32s3-elf/esp-2021r2-8.4.0/xtensa-esp32s3-elf/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld: esp-idf/pthread/libpthread.a(pthread_rwlock.c.obj): in function `pthread_rwlock_destroy':
          /home/user/.espressif/esp-idf-master/components/pthread/pthread_rwlock.c:103: multiple definition of `pthread_rwlock_destroy'; /home/user/projects/rust-esp32-std-demo/target/xtensa-esp32s3-espidf/release/deps/libesp_idf_sys-033449d0383ac0dd.rlib(esp_idf_sys-033449d0383ac0dd.esp_idf_sys.208449cf-cgu.8.rcgu.o):/home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/esp-idf-sys-0.24.4/src/pthread_rwlock.rs:54: first defined here
          /home/user/.espressif/tools/xtensa-esp32s3-elf/esp-2021r2-8.4.0/xtensa-esp32s3-elf/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld: esp-idf/pthread/libpthread.a(pthread_rwlock.c.obj): in function `pthread_rwlock_rdlock':
          /home/user/.espressif/esp-idf-master/components/pthread/pthread_rwlock.c:162: multiple definition of `pthread_rwlock_rdlock'; /home/user/projects/rust-esp32-std-demo/target/xtensa-esp32s3-espidf/release/deps/libesp_idf_sys-033449d0383ac0dd.rlib(esp_idf_sys-033449d0383ac0dd.esp_idf_sys.208449cf-cgu.8.rcgu.o):/home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/esp-idf-sys-0.24.4/src/pthread_rwlock.rs:24: first defined here
          /home/user/.espressif/tools/xtensa-esp32s3-elf/esp-2021r2-8.4.0/xtensa-esp32s3-elf/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld: esp-idf/pthread/libpthread.a(pthread_rwlock.c.obj): in function `pthread_rwlock_wrlock':
          /home/user/.espressif/esp-idf-master/components/pthread/pthread_rwlock.c:195: multiple definition of `pthread_rwlock_wrlock'; /home/user/projects/rust-esp32-std-demo/target/xtensa-esp32s3-espidf/release/deps/libesp_idf_sys-033449d0383ac0dd.rlib(esp_idf_sys-033449d0383ac0dd.esp_idf_sys.208449cf-cgu.8.rcgu.o):/home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/esp-idf-sys-0.24.4/src/pthread_rwlock.rs:36: first defined here
          /home/user/.espressif/tools/xtensa-esp32s3-elf/esp-2021r2-8.4.0/xtensa-esp32s3-elf/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld: esp-idf/pthread/libpthread.a(pthread_rwlock.c.obj): in function `pthread_rwlock_unlock':
          /home/user/.espressif/esp-idf-master/components/pthread/pthread_rwlock.c:223: multiple definition of `pthread_rwlock_unlock'; /home/user/projects/rust-esp32-std-demo/target/xtensa-esp32s3-espidf/release/deps/libesp_idf_sys-033449d0383ac0dd.rlib(esp_idf_sys-033449d0383ac0dd.esp_idf_sys.208449cf-cgu.8.rcgu.o):/home/user/.cargo/registry/src/github.com-1ecc6299db9ec823/esp-idf-sys-0.24.4/src/pthread_rwlock.rs:48: first defined here
          collect2: error: ld returned 1 exit status



error: could not compile `rust-esp32-std-demo` due to previous error
```

Reverting to the previous before the above commit, ie https://github.com/espressif/esp-idf/commit/cd97e95a125c00dd41375d990e99db0a0618a1ab on the release/v4.4 branch, results in a successful build.

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -17,7 +17,7 @@ mod start;
 
 // (Temporary code) ESP-IDF does not (yet) have a pthread rwlock implementation, which is required by STD
 // We provide a quick and very hacky implementation here
-#[cfg(feature = "std")]
+#[cfg(all(feature = "std", esp_idf_version = "4.3"))]
 mod pthread_rwlock;
 
 // (Temporary code) ESP-IDF current stable version (4.3) has atomics for ESP32S2, but not for ESP32C3
@@ -30,10 +30,13 @@ mod atomics_esp32c3;
 ///
 /// This function will become no-op once ESP-IDF V4.4 is released
 pub fn link_patches() -> (*mut c_types::c_void, *mut c_types::c_void) {
-    #[cfg(feature = "std")]
+    #[cfg(all(feature = "std", esp_idf_version = "4.3"))]
     let rwlock = pthread_rwlock::link_patches();
 
-    #[cfg(not(feature = "std"))]
+    #[cfg(any(
+        not(feature = "std"),
+        not(all(feature = "std", esp_idf_version = "4.3"))
+    ))]
     let rwlock = core::ptr::null_mut();
 
     #[cfg(all(esp32c3, esp_idf_version = "4.3"))]

